### DISCUSSION TITLE
Hue devices showing different effect options

### DISCUSSION THREAD

**Discussion by kris-anderson** [Koenkk/zigbee2mqtt#16108] [Devices]
I have some bulbs here that support effects like candle and colorloop:
https://www.zigbee2mqtt.io/devices/9290024717.html

```
The possible values are: blink, breathe, okay, channel_change, candle, fireplace, colorloop, finish_effect, stop_effect, stop_hue_effect.
```

I also have other hue devices like these that don't support those effects:
https://www.zigbee2mqtt.io/devices/1743930P7.html

```
The possible values are: blink, breathe, okay, channel_change, finish_effect, stop_effect.
```

What determines if an effect is supported or not? It would be awesome if I could get my outside lamps (the second link) to look like they're candles, but that effect isn't available in zigbee2mqtt. 

Trying to manually assign the effect via Home Assistant also has no effect. The documentation shows that the effect I'm looking for isn't available on this specific device, but is that something hard coded into zigbee2mqtt? Or is this dynamic and Zigbee2mqtt is simply reporting what effects are available from the lights themselves?

**Comment by sjorge** [comment]
The new effects were only enabled on devices that have been verified that they work. 
e.g. https://github.com/Koenkk/zigbee-herdsman-converters/pull/5267/files

You can just edit devices/philips.js or add an external converter and test it on your outdoor light, if it works a PR can be made so it will be included in the next release.

**Reply by kris-anderson** [reply]
Thank you!

I was able to make that change and test to verify that those Hue Wall Lanterns do support the effects and work well!

I created a PR here:
https://github.com/Koenkk/zigbee-herdsman-converters/pull/5311

I don't know if this is needed, or correct: `colorTempRange: [153, 500]`. That's what many of them had, although some others were set to `{colorTempRange: [153, 454]}`. Let me know if there's a way for me to test those values and I can update the PR.

**Reply by sjorge** [reply]
Yes, please also correct the color temp range

**Reply by kris-anderson** [reply]
Just to confirm, I should change what I had in my PR and use `{colorTempRange: [153, 454]}` instead?

**Reply by sjorge** [reply]
https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html#_3-adding-converter-s-for-your-device

You can ask the device for the correct one to be 100% sure.

**Reply by kris-anderson** [reply]
Perfect, thank you!

We should be good then. My device reported the same values that are in my PR:
`Read result of 'lightingColorCtrl': {"colorTempPhysicalMin":153,"colorTempPhysicalMax":500}`

**Reply by sjorge** [reply]
Then you can leave it as 153-500, I think generally color lamps have 153-500 and white spectrum only have 153-454, but history a lot of the white spectrum have 153-500 set to be on the safe side as nobody owned all those bulbs when this was introduced.
